Cargo.toml

```diff
@@ -18,3 +18,52 @@ codespan-reporting = "0.11"
 [dev-dependencies]
 indoc = "2.0"
 strip-ansi-escapes = "0.2"
+
+[lints.rust]
+# Lint groups are set to warn so new lints are used as they become available
+future_incompatible = { level = "warn", priority = -1 }
+let_underscore = { level = "warn", priority = -1 }
+nonstandard-style = { level = "warn", priority = -1 }
+rust_2018_compatibility = { level = "warn", priority = -1 }
+rust_2018_idioms = { level = "warn", priority = -1 }
+rust_2021_compatibility = { level = "warn", priority = -1 }
+unused = { level = "warn", priority = -1 }
+warnings = { level = "warn", priority = -1 }
+
+# 2024 compatibility is allow for now and will be fixed in a near-future PR
+rust_2024_compatibility = { level = "allow", priority = -2 }
+
+# We also warn on a set of individual lints that are ont included in any group
+dead_code = "warn"
+trivial_casts = "warn"
+trivial_numeric_casts = "warn"
+unsafe_code = "warn"
+unused_import_braces = "warn"
+unused_lifetimes = "warn"
+unused_macro_rules = "warn"
+unused_qualifications = "warn"
+
+# These lints are allowed, but we want to deny them over time
+async_fn_in_trait = "allow"
```

Comment:
where are those values coming from? couldn't we use the default?

Reply:
The rustc lints above are defined in these three sets:

https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html
https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html
https://doc.rust-lang.org/rustc/lints/listing/deny-by-default.html

The only rustc lint in the set above that's `warn` by default is `dead_code`.
So yea that one could be removed.